controllers/external-accounts.js

```diff
@@ -69,6 +72,17 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
+    const isUnverifiedRoleRemoved = await removeDiscordRoleUtils.removeDiscordRole(
+      req.userData,
+      attributes.discordId,
+      undefined,
+      DISCORD_ROLES.UNVERIFIED
+    );
+
+    if (!isUnverifiedRoleRemoved) {
+      return res.status(500).json({ message: "Role deletion failed. Please contact admin" });
+    }
+
```

Comment:
We can improve the error messages by providing detailed messages that help debug the issue in production.

Reply:
I've added the changes. Can you please check it?